### ISSUE TITLE
Memory leak occurs when using C# programming in the `_input` function.

### ISSUE THREAD

**Issue by xixipig19960424** [godotengine/godot#99347]
### Tested versions

Reproducible in 4.3.stable.mono

### System information

Godot v4.3.stable.mono - Windows 10.0.19045 - Vulkan (Forward+) - dedicated NVIDIA GeForce RTX 3060 Laptop GPU (NVIDIA; 32.0.15.6614) - 11th Gen Intel(R) Core(TM) i7-11800H @ 2.30GHz (16 Threads)

### Issue description

Memory leak occurs when using C# programming in the `_input` function.  

1. The issue exists across C# SDK versions 6/7/8.  
2. The issue is observed on RTX 3060/4060 and Intel Iris(R) Xe GPUs.  
3. This issue does not occur with GDScript.  

*PS: For detailed information, please refer to the attached.  [video](https://youtu.be/oPQcexuXTA8)
**Conclusion**: The issue is unrelated to the C# SDK or GPU drivers. 
**Suspected cause**: a problem in the connection between C# and C++ within Godot.

### Steps to reproduce

1-New secen Node2D
2-add script c#:
```csharp
  using Godot;
  using System;
  
  public partial class Main : Node2D
  {
	  public override void _Input(InputEvent @event)
	  {

	  }
  }
```
3-run and any method to enable window input.
4-notice Debugger -> Monitors -> Memory -> Static

### Minimal reproduction project (MRP)

N/A

**Comment by Zireael07** [comment]
I believe this is already reported and related to how GC works?

**Comment by xixipig19960424** [comment]
@Zireael07 
I tried for a long time, and you are right! It does release memory at a certain point.

**Comment by xixipig19960424** [comment]
closes #99347
